### PR TITLE
weekly graph

### PR DESCRIPTION
**Pull request by jdlopez114**
This is actually weekly graph and not monthly like the branch name.

### CODE REVIEW

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/fragments/ContactStatsFragment.java [hunk 1]
```diff
@@ -51,10 +46,15 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
 
         MonthlyGraph monthlyGraph = new MonthlyGraph(getContext(), lineGraph, lstSms);
         monthlyGraph.showMonthlyGraph();
+
+//        WeeklyGraph weeklyGraph = new WeeklyGraph(getContext(), lineGraph, lstSms);
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/fragments/ContactStatsFragment.java:50]: Why is this commented out?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+        this.context = context;
+        this.lineGraph = lineGraph;
+        this.lstSms = lstSms;
+    }
+
+    public void showWeeklyGraph() {
+
+        getLineGraphValues(lstSms);
+        getHigehstValueForYaxis();
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:63]: getHighest

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    private static final int FRI = 6;
+    private static final int SAT = 7;
+    private static final int DEFAULT_VALUE = 0;
+    private final String TAG = "sms";
+    private Context context;
+    private int highestValue;
+    private LineChartView lineGraph;
+    private float[] receivedValues;
+    private float[] sentValues;
+    private ArrayList<Sms> lstSms;
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:51]: listSms

Don't worry, we have plenty of vowels ;)

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    private void getLineGraphValues(ArrayList<Sms> lstSms) {
+
+        receivedValues = setValues(getDailyReceived(lstSms));
+        sentValues = setValues(getMonthlySent(lstSms));
+    }
+
+    private float[] setValues(TreeMap<Integer, Integer> numberOfTexts) {
+
+        ArrayList<Float> list = new ArrayList<Float>();
+        for (Map.Entry<Integer, Integer> entry : numberOfTexts.entrySet()) {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:76]: you can use numberOfTexts.values() if all you want are the values

https://docs.oracle.com/javase/7/docs/api/java/util/TreeMap.html#values()

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+        loadGraph();
+    }
+
+    private void getLineGraphValues(ArrayList<Sms> lstSms) {
+
+        receivedValues = setValues(getDailyReceived(lstSms));
+        sentValues = setValues(getMonthlySent(lstSms));
+    }
+
+    private float[] setValues(TreeMap<Integer, Integer> numberOfTexts) {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:73]: It seems like the only purpose of this function is to convert all the values from ints to floats?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+
+            Float value = entry.getValue().floatValue();
+            list.add(value);
+        }
+        return convertFloats(list);
+    }
+
+    private TreeMap<Integer, Integer> getDailyReceived(ArrayList<Sms> texts) {
+
+        TreeMap<Integer, Integer> monthlyReceived = setUpDailyTextMap();
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:86]: Should this be weeklyReceived?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+    private void getHigehstValueForYaxis() {
+
+        highestValue = getYValue(sentValues, receivedValues);
+    }
+
+    synchronized private int getYValue(float[] sentValues, float[] receivedValues) {
+        int maxSent = findMaximumValue(sentValues);
+        int maxReceived = findMaximumValue(receivedValues);
+        if (maxSent > maxReceived) {
+            return highestValue = (int) getRound(maxSent);
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:154]: You should either return the number or set it to highestValue. You don't need to do both.

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+        for (int i = 1; i < inputArray.length; i++) {
+
+            if (inputArray[i] > maxValue) {
+                maxValue = Math.round(inputArray[i]);
+            }
+        }
+        return (int) maxValue;
+    }
+
+    private long getRound(int input) {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:172]: Just return an int so you don't have to cast it later?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+    }
+
+    private static int findMaximumValue(float[] inputArray) {
+
+        float maxValue = inputArray[0];
+
+        for (int i = 1; i < inputArray.length; i++) {
+
+            if (inputArray[i] > maxValue) {
+                maxValue = Math.round(inputArray[i]);
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:166]: Either maxValue should be an int or you don't need to round. Just round at the end.

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+                context.getString(R.string.sun),
+                context.getString(R.string.mon),
+                context.getString(R.string.tue),
+                context.getString(R.string.wed),
+                context.getString(R.string.thu),
+                context.getString(R.string.fri),
+                context.getString(R.string.sat),
+        };
+
+        LineSet dataSet = new LineSet(xAxisLabels, receivedValues);
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:197]: Name this better.

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+        dataSet.setColor(parseColor(YELLOW_CRAYOLA))
+                .setDotsColor(parseColor(RED_ROSE_MADDER))
+                .setFill(parseColor(BLUE_SAPPHIRE))
+                .setThickness(6)
+                .beginAt(0);
+        lineGraph.addData(dataSet);
+
+        LineSet dataset = new LineSet(xAxisLabels, sentValues);
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:205]: DEFINITELY name this better. `dataSet` and `dataset`? Really? lol

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java [hunk 12]
```diff
@@ -0,0 +1,240 @@
...
+        dataset.setColor(parseColor("#b01cff"))
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:206]: Use color variables that you set earlier?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java [hunk 13]
```diff
@@ -0,0 +1,240 @@
...
+                .setDashed(new float[]{15f, 10f})
+                .setThickness(6)
+                .beginAt(0);
+        lineGraph.addData(dataset);
+    }
+
+    private void setGraphAttributes() {
+
+        //todo REVIEW I forgot why I did this but I believe it crashes if we don't set this value to 100
+        setHighestValueTo100();
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:217]: If this crashes, then your previous setHighestValue function did something wrong...

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+    protected TreeMap<Integer, Integer> doInBackground(WeeklyTask... params) {
+
+        ArrayList<Sms> listSms = params[0].listSms;
+        weeklyTexts = params[0].weeklyTexts;
+        return getSmsStats(listSms);
+    }
+
+    @Override
+    protected void onPostExecute(TreeMap<Integer, Integer> ret) {
+        super.onPostExecute(ret);
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java:33]: If all you do is call the superclass method, you don't need to override it.

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+    protected void onPreExecute() {
+
+        Log.i("WeeklyReceivedTask", "Getting Weekly Total Received...");
+    }
+
+    @Override
+    protected TreeMap<Integer, Integer> doInBackground(WeeklyTask... params) {
+
+        ArrayList<Sms> listSms = params[0].listSms;
+        weeklyTexts = params[0].weeklyTexts;
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java:27]: Why do you set weeklyTexts here? You overwrite it with your next call to getSmsStats()

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+        for (int i = 0; i < list.size(); i++) {
+            if (list.get(i).getType().equals("1")) {
+                receivedSms.add(list.get(i).getTime());
+            }
+        }
+        weeklyTexts = getMonthlyTexts(receivedSms);
+        return weeklyTexts;
+    }
+
+    private TreeMap<Integer, Integer> getMonthlyTexts(ArrayList<String> list) {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java:49]: getWeeklyTexts()?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+    protected void onPostExecute(TreeMap<Integer, Integer> ret) {
+        super.onPostExecute(ret);
+    }
+
+    private TreeMap<Integer, Integer> getSmsStats(ArrayList<Sms> list) {
+
+        ArrayList<String> receivedSms = new ArrayList<>();
+
+        for (int i = 0; i < list.size(); i++) {
+            if (list.get(i).getType().equals("1")) {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java:41]: "1" shouldn't be hardcoded - set it as a variable somewhere

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+        weeklyTexts = getMonthlyTexts(receivedSms);
+        return weeklyTexts;
+    }
+
+    private TreeMap<Integer, Integer> getMonthlyTexts(ArrayList<String> list) {
+
+        for (int i = 0; i < list.size(); i++) {
+
+            long lg = Long.parseLong(list.get(i));
+            DateTime juDate = new DateTime(lg);
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java:54]: what is juDate?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+            int month = juDate.getMonthOfYear();
+
+            if (weeklyTexts.containsKey(month)) {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java:57]: what if it doesn't contain the month?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+        return weeklyTexts;
+    }
+
+    private TreeMap<Integer, Integer> getMonthlyTexts(ArrayList<String> list) {
+
+        for (int i = 0; i < list.size(); i++) {
+
+            long lg = Long.parseLong(list.get(i));
+            DateTime juDate = new DateTime(lg);
+            int month = juDate.getMonthOfYear();
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java:55]: Should this be day of week?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklySent.java [hunk 21]
```diff
@@ -0,0 +1,70 @@
+package nyc.c4q.jonathancolon.inContaq.utlities.sms.weekly;
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklySent.java:1]: Same comments on this file as WeeklyReceived.java

**Review by robertjli — CHANGES_REQUESTED** [review]

**Comment by robertjli** [comment]
ping! any updates on these?

**Comment by joncolon** [comment]
The intention is there, once major bugs are cleared

**Comment by robertjli** [comment]
Cool, feel free to update this PR whenever. Just wanted to make sure you guys didn't lose this one

**Comment by joncolon** [comment]
Actually you're totally right. I didn't even realize this wasn't my branch. I'm going to go through it now.

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/MonthlyGraph.java [hunk 22]
```diff
@@ -191,14 +183,22 @@ synchronized private void loadGraph() {
...
+                context.getString(R.string.mar),
+                context.getString(R.string.apr),
+                context.getString(R.string.jun),
+                context.getString(R.string.may),
+                context.getString(R.string.jul),
+                context.getString(R.string.aug),
+                context.getString(R.string.sep),
+                context.getString(R.string.oct),
+                context.getString(R.string.nov),
+                context.getString(R.string.dec)
```
Comment (abassawo) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/MonthlyGraph.java:197]: Can also represent as string array in resources.

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java (hunk 5 above)
Comment (abassawo) [reply app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:73]: yea you could try doing something like getDailyReceived(listSms).values().toArray(); Also setValues looks like it's really a getter method.

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java (hunk 6 above)
Comment (abassawo) [reply app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:86]: any reason for the TreeMap vs any other map?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java [hunk 23]
```diff
@@ -0,0 +1,240 @@
...
+        return monthlySent;
+    }
+
+    // This converts the generic "Float" to primitive float because the graphs only uses primitives
+    private float[] convertFloats(List<Float> floats) {
+
+        float[] ret = new float[floats.size()];
+        Iterator<Float> iterator = floats.iterator();
+        for (int i = 0; i < ret.length; i++) {
+            ret[i] = iterator.next().intValue();
```
Comment (abassawo) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:125]: Do we need an iterator and iterator.next() if we're just looping from 0 to the length of ret, which is same length as floats?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+    private TreeMap<Integer, Integer> getSmsStats(ArrayList<Sms> list) {
+
+        ArrayList<String> receivedSms = new ArrayList<>();
+
+        for (int i = 0; i < list.size(); i++) {
+            if (list.get(i).getType().equals("1")) {
+                receivedSms.add(list.get(i).getTime());
+            }
+        }
+        weeklyTexts = getMonthlyTexts(receivedSms);
```
Comment (abassawo) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java:45]: which one is it?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+        return weeklyTexts;
+    }
+
+    private TreeMap<Integer, Integer> getMonthlyTexts(ArrayList<String> list) {
+
+        for (int i = 0; i < list.size(); i++) {
+
+            long lg = Long.parseLong(list.get(i));
```
Comment (abassawo) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java:53]: better name for lg. what does this represent?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklySent.java [hunk 26]
```diff
@@ -0,0 +1,70 @@
...
+    @Override
+    protected void onPreExecute() {
+        Log.i("WeeklySentTask", "Getting Weekly Total Sent...");
+    }
+
+    @Override
+    protected TreeMap<Integer, Integer> doInBackground(WeeklyTask... params) {
+
+        ArrayList<Sms> listSms = params[0].listSms;
+        weeklyTexts = params[0].weeklyTexts;
```
Comment (abassawo) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklySent.java:26]: this looks like it doesn't belong in the doInBackground...not sure though. why is this being changed in doInBackground

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklySent.java [hunk 27]
```diff
@@ -0,0 +1,70 @@
...
+        return getSmsStats(listSms);
+
+    }
+
+    @Override
+    protected void onPostExecute(TreeMap<Integer, Integer> ret) {
+        super.onPostExecute(ret);
```
Comment (abassawo) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklySent.java:33]: what else should happen after the background work is executed?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklySent.java [hunk 28]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    private TreeMap<Integer, Integer> getSmsStats(ArrayList<Sms> list) {
+
+        ArrayList<String> sentSms = new ArrayList<>();
```
Comment (abassawo) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklySent.java:38]: List<String> sentSms = new ArrayList<>(); 
get in the habit of declaring by interface.

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java (hunk 6 above)
Comment (hkim2013) [reply app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java:86]: Yes, Jon looked up that TreeMaps can sort unlike other maps. He left a comment there, but I deleted it. Should we keep the comments there?

**Pull request merged**